zulip/integrations/google/google-calendar

```diff
@@ -71,30 +83,77 @@ parser.add_argument(
-    default=CLIENT_SECRET_PATH,
     dest="client_secret_path",
 )
 parser.add_argument(
     "--tokens-file",
     help=f"The path to the file containing the tokens for the Google Calendar API. By default, the tokens file is generated at {TOKENS_PATH} after the first run.",
-    default=TOKENS_PATH,
     dest="tokens_path",
 )
 parser.add_argument(
     "-n",
     "--noauth_local_webserver",
     action="store_true",
-    default=False,
     help="The default authorization process runs a local web server, which requires a browser on the same machine. For non-interactive environments and machines without browser access, e.g., remote servers, this option allows manual authorization. The authorization URL is printed, which the user can copy into a browser, copy the resulting authorization code, and paste back into the command line.",
 )
-options = parser.parse_args()
+commandline_options = parser.parse_args()
+if commandline_options.verbose:
+    logging.getLogger().setLevel(logging.INFO)
 
-zulip_client = zulip.init_from_options(options)
+
+valid_keys = list(GoogleCalendarOptions.__dataclass_fields__.keys())
+
+
+def load_config_options(config_path: Optional[str]) -> GoogleCalendarOptions:
+    if config_path is None:
+        config_path = zulip.get_default_config_filename()
+    assert config_path is not None
+    if not os.path.exists(config_path):
+        logging.info("No config file found at %s", config_path)
+        return GoogleCalendarOptions()
+
+    logging.info("Loading Google Calendar configuration from %s", config_path)
+    config = ConfigParser()
+    try:
+        config.read(config_path)
+    except Exception:
+        logging.exception("Error reading config file %s", config_path)
+
+    section = "google-calendar"
+    config_values = {}
+    if section in config:
+        for key, value in config[section].items():
+            if key in valid_keys:
+                expected_type = GoogleCalendarOptions.__annotations__[key]
+                config_values[key] = True if expected_type == bool else expected_type(value)
+                logging.info("Setting key: %s to %s", key, config_values[key])
+            else:
+                logging.warning(
+                    "Unknown key %s in section %s of config file %s", key, section, config_path
+                )
+    return GoogleCalendarOptions(**config_values)
+
+
+def update_calendar_options_from_commandline_args(
+    calendar_options: GoogleCalendarOptions, commandline_options: argparse.Namespace
+) -> None:
+    for key, value in commandline_options.__dict__.items():
+        # Boolean arguments (store-true) have a default value of False when not passed in.
+        # So, we ignore them, to prevent overwriting the config file option that is set.
+        if key in valid_keys and value is not None and value is not False:
+            setattr(calendar_options, key, value)
```

Comment:
Probably these two functions will want to eventually move to common code that other integrations can use too. But this is fine for this PR.